### DISCUSSION TITLE
Request Complete guide for Column Encryption

### DISCUSSION THREAD

**Discussion by mzvdev** [supabase/supabase#20390] [Feature Requests]
Hi,

Is it possible to publish a more complete guide and tutorial for column encryption:

1. Where're stored the keys?
2. How backup the keys?
3. How reset the keys?
4. How rotate the keys?
5. In case of error how we recover SupaBase project with backuped keys?
6. Which method of encryption to use with custom RLS which allow create/update/insert for members of an organization ?
    1. **One Key ID for the Entire Column**
    2. **One Key ID per Row**
    3. **One Key ID per Row with Nonce Support**
    4. **How works :One Key ID per Row with Associated Data?**
7. How it works with migration sql files

**Comment by roo12312** [comment]
even i need answers
